pkg/discovery/dtofactory/pod_entity_dto_builder.go

```diff
@@ -285,6 +288,26 @@ func (builder *podEntityDTOBuilder) buildDTOs(pods []*api.Pod, resCommTypeSold,
 	return result
 }
 
+func (builder *podEntityDTOBuilder) isContaierMetricsAvailable(pod *api.Pod) bool {
+	// We consider that the metrics for a pods containers are available until we don't
+	// explicitly see an availability metrics set to false.
+	isAvailable := true
+	entityKey := util.PodMetricIdAPI(pod)
+	ownerMetricId := metrics.GenerateEntityStateMetricUID(metrics.PodType, entityKey, metrics.MetricsAvailability)
+	availabilityMetric, err := builder.metricsSink.GetMetric(ownerMetricId)
+	if err != nil {
+		glog.Warningf("Error getting %s from metrics sink for pod %s --> %v", metrics.MetricsAvailability, entityKey, err)
+	} else {
+		availabilityMetricValue := availabilityMetric.GetValue()
+		ok := false
+		isAvailable, ok = availabilityMetricValue.(bool)
+		if !ok {
+			glog.Warningf("Error getting %s from metrics sink for pod %s. Wrong type: %T, Expected: bool.", metrics.MetricsAvailability, entityKey, availabilityMetricValue)
+		}
+	}
```

Comment:
I don't know of a reason this might happen. As you said, if this happens something is seriously broken (the metrics is not written to sink or what is written is not the right type). However, my thought behind doing this was to ensure, the behaviour of code to remain similar to what it was without this change even in the event of such issues, i.e. we treated the node to be available (offcourse, subject to other already existing checks). 

Please let me know if that does not still seem right.

Reply:
Thanks @irfanurrehman, now that you explained your thoughts, I don't have a strong opinion one way or the other. I am fine with the current approach.